src/main/webapp/new/app/play/game-map/game-map.component.ts

```diff
@@ -83,9 +83,16 @@ export class GameMapComponent implements OnInit {
     }
 
     private _subscribeOnMarkingChanging() {
-        this._marking.onUpdate('hexes', () => this._drawMapMarkingService.updateByType(this._canvas, 'hexes', 'hex'));
-        this._marking.onUpdate('nodes', () => this._drawMapMarkingService.updateByType(this._canvas, 'nodes', 'node'));
-        this._marking.onUpdate('edges', () => this._drawMapMarkingService.updateByType(this._canvas, 'edges', 'edge'));
+        let types = [
+            ['hexes', 'hex'],
+            ['nodes', 'node'],
+            ['edges', 'edge']
+        ];
+
+        types.forEach(type => {
+            this._drawMapMarkingService.updateByType(this._canvas, type[0], type[1]);
```

Comment:
If you update the page when you have mandatory AA (e.g BUILD_ROAD) then setting of marking happens before subscription on marking change (because game-map is created after parent 'play.component')